pkg/sql/doctor/doctor_test.go

```diff
@@ -413,6 +420,38 @@ func TestExamineDescriptors(t *testing.T) {
+			// foreign key references.
+			//
+			//  CREATE TABLE a (i INT8 PRIMARY KEY);
+			//  CREATE TABLE b (i INT8 PRIMARY KEY, j INT8 REFERENCES a (i));
+			//  CREATE TABLE c (i INT8 PRIMARY KEY REFERENCES b (i));
+			//
+			descTable: doctor.DescriptorTable{
+				{ID: 57, DescBytes: fromHex(t, "12240a0964656661756c74646210391a150a090a0561646d696e10020a080a04726f6f741002")},
+				{ID: 58, DescBytes: fromHex(t, "0aed010a0161183a203928043a0a08c2a7a5b1b7b9c8b81642170a016910011a0c08011040180030005014600020003000480252560a077072696d61727910011801220169300140004a10080010001a00200028003000380040005210083510021a00200028003000380040005a007a020800800100880100900100980100a20106080012001800a8010060026a150a090a0561646d696e10020a080a04726f6f741002800101880103980100b201120a077072696d61727910001a016920012800b80101c20100e80100f2010408001200f801008002009202009a0200b20209726573746f72696e67b80200c00200")},
+				{ID: 59, DescBytes: fromHex(t, "0aed020a0162183b203928053a0a08abc5d096d7b9c8b81642170a016910011a0c0801104018003000501460002000300042170a016a10021a0c08011040180030005014600020013000480352560a077072696d61727910011801220169300140004a10080010001a00200028003000380040005210083610011a00200028003000380040005a007a020800800100880100900100980100a20106080012001800a801005a600a17625f6175746f5f696e6465785f666b5f6a5f7265665f611002180022016a3002380140004a1a083410011a0a666b5f6a5f7265665f61200028013000380040005a007a020800800100880100900100980100a20106080012001800a8010060036a150a090a0561646d696e10020a080a04726f6f741002800101880103980100b201170a077072696d61727910001a01691a016a200120022802b80101c20100e80100f2010408001200f801008002009202009a0200b20209726573746f72696e67b80200c00200")},
+				{ID: 60, DescBytes: fromHex(t, "0ae5010a0163183c203928043a0a08fbf39aacd7b9c8b81642170a016910011a0c080110401800300050146000200030004802524e0a077072696d61727910011801220169300140004a1a083510011a0a666b5f695f7265665f62200028013000380040005a007a020800800100880100900100980100a20106080012001800a8010060026a150a090a0561646d696e10020a080a04726f6f741002800101880103980100b201120a077072696d61727910001a016920012800b80101c20100e80100f2010408001200f801008002009202009a0200b20209726573746f72696e67b80200c00200")},
+			},
+			namespaceTable: doctor.NamespaceTable{
+				{NameInfo: descpb.NameInfo{ParentID: 0, ParentSchemaID: 0, Name: "defaultdb"}, ID: 57},
+				{NameInfo: descpb.NameInfo{ParentID: 57, ParentSchemaID: 29, Name: "a"}, ID: 58},
+				{NameInfo: descpb.NameInfo{ParentID: 57, ParentSchemaID: 29, Name: "b"}, ID: 59},
+				{NameInfo: descpb.NameInfo{ParentID: 57, ParentSchemaID: 29, Name: "c"}, ID: 60},
+			},
+			expected: `Examining 4 descriptors and 4 namespace entries...
+  ParentID  57, ParentSchemaID 29: relation "a" (58): referenced table ID 53: descriptor not found
+  ParentID  57, ParentSchemaID 29: relation "b" (59): referenced table ID 52: descriptor not found
+  ParentID  57, ParentSchemaID 29: relation "c" (60): referenced table ID 53: descriptor not found
```

Comment:
I'm happy with this test case as is, but may I suggest adding a valid table descriptor at ID `52` but that's missing a backreference to table 59? It'd be a cheap way to increase coverage.